### DISCUSSION TITLE
SNAT ICMP to enable ping

### DISCUSSION THREAD

**Discussion by strxno** [fakemanhk/openwrt-jp-ipoe#6] [Ideas]
Can someone else test that this enables ICMP ping? I got it to work for a little while and after I updated the rule it stopped working completely again. The map.sh file that comes with the map plugin is trying to set it incorrectly by providing ports to the rule, but this should set it correctly.

[map.sh.txt](https://github.com/fakemanhk/openwrt-jp-ipoe/files/10788331/map.sh.txt)

**Comment by fakemanhk** [comment]
Just tested, the PING still not working......

**Reply by strxno** [reply]
I found this site which shows how to map nftables ICMP via IPv4 using ports. I have no idea how this works and will test later today.

https://mari.sh/2019/11/apu2-alpine-linux-and-v6plus/

**Reply by strxno** [reply]
Yep it worked. Adding the rule to nftables immediately allowed ping to work. I'll have to test and see if I can add this to the script.

**Comment by cmspam** [comment]
You can see my last post in the openwrt thread, for an idea how to modify the map.sh to work.

I got it working but feel it's a somewhat hackish way. Just modify nftables directly from the map.sh script.

https://forum.openwrt.org/t/a-guide-for-japan-ntt-users-on-ipoe-ipv6-setting-map-e-only/150587/4
